darts/models/forecasting/regression_model.py

```diff
@@ -324,19 +328,36 @@ def _create_lagged_data(
         lags_past_covariates = self.lags.get("past")
         lags_future_covariates = self.lags.get("future")
 
-        training_samples, training_labels, _ = _create_lagged_data(
-            target_series=target_series,
-            output_chunk_length=self.output_chunk_length,
-            past_covariates=past_covariates,
-            future_covariates=future_covariates,
-            lags=lags,
-            lags_past_covariates=lags_past_covariates,
-            lags_future_covariates=lags_future_covariates,
-            max_samples_per_ts=max_samples_per_ts,
-            multi_models=self.multi_models,
-        )
+        # ensure list of TimeSeries format
+        if isinstance(target_series, TimeSeries):
+            target_series = [target_series]
+            past_covariates = [past_covariates] if past_covariates else None
+            future_covariates = [future_covariates] if future_covariates else None
+
+        training_samples, training_labels = [], []
+        for i, target in enumerate(target_series):
+            samples_i, labels_i, _ = create_lagged_training_data(
```

Comment:
Definitely - I steered clear of that for now since, admittedly, I haven't thoroughly looked through the `prediction` code and because I'm cognizant of the fact that this PR has become pretty big. With that being said, I'm definitely keen to have a go doing this if this PR is merged.